mantidproject/mantid
# Allow updates for facilities.xml  through automatic downloading

The facilities.xml file is downloaded together with the instrument definitions. However it was only ever loaded from the install directory.

This update makes it load from the same list of directories as the instrument files.

**In order of preference:**

1. The download directory (win %appdata%\mantidproject\mantid\instrument) (linux $home/.mantid/instrument )
2. The user instrument area (don't worry too much about this one / only available on linux)
3. The install directory/instrument

**To test:**

1. This is assuming you have built a developer version of Mantid (automatic downloading is off).
1. Go to the download directory, and if there is a Facilities.xml file delete it.
1. Start up Mantid, it should start fine, and have facilities and instruments in the drop down lists of the first time use dialog of the help menu. These will have loaded from the install directory (you could edit the file to check this if you wanted).
1. Delete the github.json file from the download directory, this will force an update on running DownloadInstrument.
1. Run the DownloadInstrument alg.
1. As the algorithm completes you should have the github.json file recreated, and possibly other .xml files downloaded, if you are lucky that will include a Facilities.xml file
  1. If you did get a new Facilities.xml file then you should also have got a Notice level log telling you it will be loaded on the next restart of Mantid.
  1. If you did not then copy a facilities.xml file into this directory from the installation directory, 
1. Edit the facilities.xml file in the download directory to change an instrument name slightly.
1. Close and restart mantidplot
1. The edited instrument name should be visible in the first use dialog.

**Clean up:**

1. Delete the edited Facilities.xml file from the download directory

Fixes #18444

#### RELEASE NOTES
Either edit the file in docs/source/release/framework.xml

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
